src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSourceCredentials.cs

```diff
@@ -46,13 +47,14 @@ NetworkCredential ICredentials.GetCredential(Uri uri, string authType)
         {
             // Credentials may change during this call so keep a local copy.
             var currentCredentials = Credentials;
-
             NetworkCredential result = null;
 
             if (currentCredentials == null)
             {
                 // This is used to match the value of HttpClientHandler.UseDefaultCredentials = true
-                result = CredentialCache.DefaultNetworkCredentials;
+                result = PreviewFeatureSettings.DefaultCredentialsAfterCredentialProviders
+                    ? null
+                    : CredentialCache.DefaultNetworkCredentials;
```

Comment:
By analogy with `HttpClientHandler.UseDefaultCredentials` I'd add a static property to `HttpHandlerResourceV3` and would plumb it from there resource => handler => credentials.
Does it make sense?